src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/PackageSourceMappingUtility.cs

```diff
@@ -32,8 +32,8 @@ internal static class PackageSourceMappingUtility
             List<string> addedPackageIdsWithoutExistingMappings = new();
             foreach (string addedPackageId in addedPackageIds)
             {
-                IReadOnlyList<string>? configuredSource = packageSourceMapping.GetConfiguredPackageSources(addedPackageId);
-                if (configuredSource is null || configuredSource.Count == 0)
+                IReadOnlyList<string> configuredSource = packageSourceMapping.GetConfiguredPackageSources(addedPackageId);
```

Comment:
I think this is incorrect. Other places calling this method check for null as well. Looking at the implementation, it seems like nulls are possible.

Reply:
The `SearchTree` returns an empty list
https://github.com/NuGet/NuGet.Client/blob/1fbe83fe674911a7f1289dad525be6ff73b7d2e5/src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/SearchTree.cs#L100

Reply:
After I read the code, I realized that you changed that as well. Resolving this thread.